tests/test_markers.py

```diff
@@ -388,12 +387,38 @@ def test_extra_str_normalization(self):
         assert str(Marker(rhs)) == f'extra == "{normalized_name}"'
 
     def test_python_full_version_untagged_user_provided(self):
-        """A user-provided python_full_version ending with a + fails to parse."""
-        with pytest.raises(InvalidVersion):
-            Marker("python_full_version < '3.12'").evaluate(
-                {"python_full_version": "3.11.1+"}
-            )
+        """A user-provided python_full_version ending with a + uses Python behaviour."""
+        env = {"python_full_version": "3.11.1+"}
+        assert Marker("python_full_version < '3.12'").evaluate(env)
```

Comment:
With this change, this fallbacks to Python comparison semantics so it technically works now.

cc @sbidoul for visibility that you're okay with this change.